File: src/org/cerion/symcalc/expression/BoolExpr.java
package org.cerion.symcalc.expression;

public class BoolExpr extends Expr {

	public static final BoolExpr TRUE = new BoolExpr(true);
	public static final BoolExpr FALSE = new BoolExpr(false);
	
	public BoolExpr(boolean value)
	{
		setValue(value);
	}
	
	public boolean value() {
		return (boolean)this.getValue();
	}

	@Override
	public boolean equals(Expr e) {
		if(e.isBool()) {
			BoolExpr be = (BoolExpr) e;
			return be.value() == value();
		}

		return false;
	}

	@Override
	public String toString() {
		
		return (value() ? "True" : "False");
	}

	@Override
	public void show(int i) {
    	indent(i,"BoolExpr = " + toString());
	}

	@Override
	protected Expr evaluate() {
		return this;
	}

	@Override
	public ExprType getType() {
		return ExprType.BOOL;
	}

}

File: src/org/cerion/symcalc/expression/BoolExpr.kt
package org.cerion.symcalc.expression

class BoolExpr(value: Boolean) : Expr() {

    override val type: ExprType
        get() = ExprType.BOOL

    companion object {
        @JvmField val TRUE = BoolExpr(true)
        @JvmField val FALSE = BoolExpr(false)
    }

    init {
        this.value = value
    }

    fun value(): Boolean  = this.value as Boolean

    override fun equals(e: Expr): Boolean {
        if (e.isBool)
            return e.asBool().value() == value()

        return false
    }

    override fun toString(): String = if (value()) "True" else "False"
    override fun treeForm(i: Int) = indent(i, "BoolExpr = " + toString())
    override fun evaluate(): Expr = this
}
